### PR TITLE
TD-13 Changed kernel.root_dir to kernel_project_dir

### PR DESCRIPTION
**Pull request by fred-jan**

### CODE REVIEW

File: Command/AbstractCommand.php [hunk 1]
```diff
@@ -230,7 +230,7 @@ protected function copySchemas(KernelInterface $kernel, $cacheDir)
             $filesystem->mkdir($cacheDir);
         }
 
-        $base = ltrim(realpath($kernel->getRootDir().'/..'), DIRECTORY_SEPARATOR);
+        $base = ltrim(realpath($kernel->getProjectDir()), DIRECTORY_SEPARATOR);
```
Comment (gertjuhh) [inline Command/AbstractCommand.php]:
```suggestion
        $base = \ltrim(\realpath($kernel->getProjectDir()), DIRECTORY_SEPARATOR);
```

File: Command/AbstractCommand.php [hunk 2]
```diff
@@ -339,7 +339,10 @@ protected function getSchemasFromBundle(BundleInterface $bundle)
      */
     protected function getRelativeFileName(\SplFileInfo $file)
     {
-        return substr(str_replace(realpath($this->getContainer()->getParameter('kernel.root_dir') . '/../'), '', $file), 1);
+        return \substr(
+            \str_replace(realpath($this->getContainer()->getParameter('kernel.project_dir').'/'), '', $file),
```
Comment (gertjuhh) [inline Command/AbstractCommand.php]:
```suggestion
            \str_replace(\realpath($this->getContainer()->getParameter('kernel.project_dir').'/'), '', $file),
```

File: Command/AbstractCommand.php [hunk 3]
```diff
@@ -652,8 +655,8 @@ private function getPhingArguments(KernelInterface $kernel, $workingDirectory, $
         $properties = array_merge(array(
             'propel.database'           => 'mysql',
             'project.dir'               => $workingDirectory,
-            'propel.output.dir'         => $kernel->getRootDir().'/propel',
-            'propel.php.dir'            => $kernel->getRootDir().'/..',
+            'propel.output.dir'         => $kernel->getProjectDir().'/var/propel',
```
Comment (gertjuhh) [inline Command/AbstractCommand.php:658]: Is this change in location intentional?
`./app/propel` --> `./var/propel`

File: Command/FixturesDumpCommand.php [hunk 4]
```diff
@@ -61,7 +61,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         list($name, $defaultConfig) = $this->getConnection($input, $output);
         $fixtureDir = $input->getOption('dir') ? $input->getOption('dir') : $this->defaultFixturesDir;
 
-        $path = realpath($this->getApplication()->getKernel()->getRootDir() . '/../') . '/' . $fixtureDir;
+        $path = realpath($this->getApplication()->getKernel()->getProjectDir() . '/') . '/' . $fixtureDir;
```
Comment (gertjuhh) [inline Command/FixturesDumpCommand.php]:
```suggestion
        $path = \realpath($this->getApplication()->getKernel()->getProjectDir() . '/') . '/' . $fixtureDir;
```

File: Command/FixturesLoadCommand.php [hunk 5]
```diff
@@ -120,7 +120,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         if (null !== $this->bundle) {
             $this->absoluteFixturesPath = $this->getFixturesPath($this->bundle);
         } else {
-            $this->absoluteFixturesPath = realpath($this->getApplication()->getKernel()->getRootDir() . '/../' . $input->getOption('dir'));
+            $this->absoluteFixturesPath = realpath($this->getApplication()->getKernel()->getProjectDir() . '/' . $input->getOption('dir'));
```
Comment (gertjuhh) [inline Command/FixturesLoadCommand.php]:
```suggestion
            $this->absoluteFixturesPath = \realpath($this->getApplication()->getKernel()->getProjectDir() . '/' . $input->getOption('dir'));
```

File: Command/FixturesLoadCommand.php [hunk 6]
```diff
@@ -204,7 +204,7 @@ protected function loadFixtures(InputInterface $input, OutputInterface $output,
      */
     protected function loadSqlFixtures(InputInterface $input, OutputInterface $output)
     {
-        $tmpdir = $this->getApplication()->getKernel()->getRootDir() . '/cache/propel';
+        $tmpdir = $this->getApplication()->getKernel()->getProjectDir() . '/var/cache/propel';
```
Comment (gertjuhh) [inline Command/FixturesLoadCommand.php:207]: Is this change in location intentional?
`./app/cache/propel` --> `./var/cache/propel`

File: Command/GraphvizGenerateCommand.php [hunk 7]
```diff
@@ -43,7 +43,7 @@ protected function configure()
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $dest = $this->getApplication()->getKernel()->getRootDir() . '/propel/graph/';
+        $dest = $this->getApplication()->getKernel()->getProjectDir() . '/var/propel/graph/';
```
Comment (gertjuhh) [inline Command/GraphvizGenerateCommand.php:46]: Is this change in location intentional?
`./app/propel/graph` --> `./var/propel/graph`

File: Command/ReverseCommand.php [hunk 8]
```diff
@@ -63,7 +63,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $filesystem = new Filesystem();
             $generated  = $this->getCacheDir().'/schema.xml';
             $filename   = $name . '_reversed_schema.xml';
-            $destFile   = $this->getApplication()->getKernel()->getRootDir() . '/propel/generated-schemas/' . $filename;
+            $destFile   = $this->getApplication()->getKernel()->getProjectDir() . '/var/propel/generated-schemas/' . $filename;
```
Comment (gertjuhh) [inline Command/ReverseCommand.php:66]: Is this change in location intentional?
`./app/propel/generated-schemas/` --> `./var/propel/generated-schemas/`

File: DependencyInjection/PropelExtension.php [hunk 9]
```diff
@@ -35,10 +35,10 @@ public function load(array $configs, ContainerBuilder $container)
         $config = $processor->processConfiguration($configuration, $configs);
 
         // Composer
-        if (file_exists($propelPath = $container->getParameter('kernel.root_dir') . '/../vendor/propel/propel1')) {
+        if (file_exists($propelPath = $container->getParameter('kernel.project_dir') . '/vendor/propel/propel1')) {
```
Comment (gertjuhh) [inline DependencyInjection/PropelExtension.php]:
```suggestion
        if (\file_exists($propelPath = $container->getParameter('kernel.project_dir') . '/vendor/propel/propel1')) {
```

File: DependencyInjection/PropelExtension.php [hunk 10]
```diff
@@ -35,10 +35,10 @@ public function load(array $configs, ContainerBuilder $container)
...
             $container->setParameter('propel.path', $propelPath);
         }
-        if (file_exists($phingPath = $container->getParameter('kernel.root_dir') . '/../vendor/phing/phing/classes')) {
+        if (file_exists($phingPath = $container->getParameter('kernel.project_dir') . '/vendor/phing/phing/classes')) {
```
Comment (gertjuhh) [inline DependencyInjection/PropelExtension.php]:
```suggestion
        if (\file_exists($phingPath = $container->getParameter('kernel.project_dir') . '/vendor/phing/phing/classes')) {
```

File: PropelBundle.php [hunk 11]
```diff
@@ -28,7 +28,7 @@ public function boot()
         require_once $this->container->getParameter('propel.path').'/runtime/lib/Propel.php';
 
         if (0 === strncasecmp(PHP_SAPI, 'cli', 3)) {
-            set_include_path($this->container->getParameter('kernel.root_dir').'/..'.PATH_SEPARATOR.
+            set_include_path($this->container->getParameter('kernel.project_dir').PATH_SEPARATOR.
```
Comment (gertjuhh) [inline PropelBundle.php]:
```suggestion
            \set_include_path($this->container->getParameter('kernel.project_dir').PATH_SEPARATOR.
```

**Review by gertjuhh — CHANGES_REQUESTED** [review]

File: Command/AbstractCommand.php (hunk 3 above)
Comment (fred-jan) [reply Command/AbstractCommand.php:658]: It is, seemed better to me to group generated files together in a subdir than in the root dir.

File: Command/FixturesLoadCommand.php (hunk 6 above)
Comment (fred-jan) [reply Command/FixturesLoadCommand.php:207]: It is, seemed better to me to group generated files together in a subdir than in the root dir.

File: Command/GraphvizGenerateCommand.php (hunk 7 above)
Comment (fred-jan) [reply Command/GraphvizGenerateCommand.php:46]: It is, seemed better to me to group generated files together in a subdir than in the root dir.

File: Command/ReverseCommand.php (hunk 8 above)
Comment (fred-jan) [reply Command/ReverseCommand.php:66]: It is, seemed better to me to group generated files together in a subdir than in the root dir.

**Review by gertjuhh — APPROVED** [review]

**Pull request merged**